src/main/java/com/exasol/parquetio/data/GenericRow.java

```diff
@@ -30,11 +47,49 @@ public static GenericRow of(final Object... values) {
         return new GenericRow(Arrays.asList(values));
     }
 
+    /**
+     * A factory method to create a new instance with {@link MessageType} schema.
+     *
+     * @param schema schema of a row
+     * @param values list of values
+     */
+    public static GenericRow of(final MessageType schema, final Object... values) {
+        return new GenericRow(schema, Arrays.asList(values));
+    }
+
+    /**
+     * Checks if a column with a given name exists.
+     *
+     * @param fieldName field name in a row
+     * @return {@code true} if column with a name exists; otherwise {@code false}
+     */
+    public boolean hasFieldName(final String fieldName) {
+        return schema.containsField(fieldName);
+    }
+
+    /**
+     * Returns list of field names.
+     *
+     * @return list of field names
+     */
+    public List<String> getFieldNames() {
+        return schema.getFields().stream().map(Type::getName).collect(Collectors.toList());
```

Comment:
Unfortunately the java stream() api is unbelivable slow.
Since this can eventually be called for each row, I think performance is more important here than the very good readablity.
One problem with `Collectors.toList()` is for example that it starts with an empty list. That requires expensive growing. 
In this case you already know the result size. By that you can do `new ArrayList(fields.size());`

Reply:
Good find! Added the change.

Later we can cache it with each row, now it is okay since we already have schema object also.